internal/controller/processor.go

```diff
@@ -38,16 +38,22 @@ type RefreshOptions struct {
 	keyVaultRefreshEnabled      bool
 	keyVaultRefreshNeeded       bool
 	updatedSentinelETags        map[acpv1.Sentinel]*azcore.ETag
+	featureFlagRefreshEnabled   bool
+	featureFlagRefreshNeeded    bool
 	ConfigMapSettingPopulated   bool
 	SecretSettingPopulated      bool
 }
 
-func (processor *AppConfigurationProviderProcessor) PopulateSettings(existingSecret *corev1.Secret) error {
+func (processor *AppConfigurationProviderProcessor) PopulateSettings(existingConfigMap *corev1.ConfigMap, existingSecret *corev1.Secret) error {
 	if err := processor.ProcessFullReconciliation(); err != nil {
 		return err
 	}
 
-	if err := processor.ProcessKeyValueRefresh(); err != nil {
+	if err := processor.ProcessFeatureFlagRefresh(existingConfigMap); err != nil {
```

Comment:
Seems if both `FeatureFlagRefresh` and `KeyValueRefresh` are triggered in same reconciliation. There would be confliction.

Reply:
When they are triggered in same reconciliation, `processor` will sequentially update parts of the feature flag and key-value. In their respective update processes, the other content will remain unchanged. What does conflict specifically refer to？